ansible/files/bootstrap/telemetry.sh

```diff
@@ -209,6 +224,62 @@ rm RRO-3.2.1-el6.x86_64.tar.gz
 cd RRO-3.2.1; sudo ./install.sh; cd $HOME
 $ANACONDA_PATH/bin/pip install rpy2
 
+# Setup metrics reporting
+SOPS_RPM=sops-3.0.4-1.x86_64.rpm
+SPARK_METRICS_CONF=/mnt/spark-metrics.properties
+touch $SPARK_METRICS_CONF
+if [ "$METRICS_PROVIDER" = "datadog" ] ; then
+    # Pull down and decrypt Datadog API key; see https://github.com/mozilla/emr-bootstrap-spark/pull/485
+    DD_DIR=$(mktemp -d)
+    aws s3 cp $TELEMETRY_CONF_BUCKET/packages/$SOPS_RPM $DD_DIR/
+    aws s3 cp $TELEMETRY_CONF_BUCKET/credentials/emr.yaml $DD_DIR/
+    sudo yum -y localinstall $DD_DIR/$SOPS_RPM
+    export DD_API_KEY=$(/usr/local/bin/sops --decrypt --extract '["datadog_agent::api_key"]' $DD_DIR/emr.yaml)
+    # Install datadog-agent; see https://app.datadoghq.com/account/settings#agent/aws
+    bash -c "$(curl -sLS https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"
+    rm -r "$DD_DIR"
+
+    # We opt not use to use https://docs.datadoghq.com/integrations/spark/
+    # Instead, we standardize on statsd metrics using Spark's StatsDSink
+    # and configure tags in the agent that will be applied by dogstatsd.
+    # This should make switching to a new metrics provider relatively straightforward.
+
+    MIN_SPARK_VERSION=2.3.0
+    SPARK_VERSION=$(describe_cluster '.Cluster.Applications[] | select(.Name=="Spark") | .Version')
+    spark_has_statsdsink() {
+        # Drop into python to take advantage of pkg_version.parse_version,
+        # which will provide more robust comparison than pure bash.
+        python <<EOF
+from sys import exit
+from pkg_resources import parse_version
+if (parse_version('${SPARK_VERSION}') < parse_version('${MIN_SPARK_VERSION}')):
+   exit(1)
+EOF
+    }
```

Comment:
This now parses based on Spark version rather than emr label, which should be more stable. Also, we now use python `pkg_version.parse_version` rather than string comparison.

Reply:
lgtm